static/controllers/favoritesController.js

```diff
@@ -1,94 +1,94 @@
-				str = "Next Service - Outbound: " + UtilityService.formatSeconds(o_pre_1) + ", " + UtilityService.formatSeconds(o_pre_2) + "<br />";
-				}
-				
-				if (i_pre_2 === null) {
-					if (i_pre_1 === null) {
-						 str += "Next Service - Inbound: There is no scheduled service to this station at this time.";
-						break;
-					}
-					else {
-						str += "Next Service - Inbound: " + UtilityService.formatSeconds(i_pre_1);
-						break;
-					}
-				}
-				str += "Inbound: " + UtilityService.formatSeconds(i_pre_1) + ", " + UtilityService.formatSeconds(i_pre_2) + ".";
-				break;
+		for (var i = 0; i<vm.stations.length; i++) {
+			if (vm.stations[i].id === station_id) {
+                str += "Outbound: "
+                    + PredictionService.makePredictionString(vm.stations[i].outbound_pre.pre_1,
+                                                            vm.stations[i].outbound_pre.pre_2)
+                    + "<br />";
+                str += "Inbound: "
+                    + PredictionService.makePredictionString(vm.stations[i].inbound_pre.pre_1,
+                                                             vm.stations[i].inbound_pre.pre_2);
+				return str;
 			}
 		}
-		
-		return str;
 	}
+
+	var refreshFavoritesData = function() {
+	    for (var i = 0; i < vm.stations.length; i++) {
+            refreshOutboundPredictions(i);
+            refreshInboundPredictions(i);
+		}
+	};
+
+	var refreshInboundPredictions = function(station_idx) {
+	    $http.get('/station/' + vm.stations[station_idx].id + '/direction/' + 1 + '/nextservice')
+			 .then(function successfulCallback(response) {
+				 vm.stations[station_idx].inbound_pre.pre_1 = response.data['prediction1'];
+				 vm.stations[station_idx].inbound_pre.pre_2 = response.data['prediction2'];
+
+			 }, function errorCallback(response) {
+		    });
+	}
+
+	var refreshOutboundPredictions = function(station_idx) {
+	    $http.get('/station/' + vm.stations[station_idx].id + '/direction/' + 0 + '/nextservice')
+			 .then(function successfulCallback(response) {
+				 vm.stations[station_idx].outbound_pre.pre_1 = response.data['prediction1'];
+				 vm.stations[station_idx].outbound_pre.pre_2 = response.data['prediction2'];
+
+			 }, function errorCallback(response) {
+		    });
+	}
+
+	var init = function() {
+		vm.stations = [];
 	
-	$scope.init = function() {
-		$scope.stations = [];
-	
-		if (FavoritesService.favoritesExist()) {
+		if (vm.favoritesExist()) {
 			for (var i = 0; i < $localStorage.favorite_stations.length; i++) {
-				$http.get('/station/' + $localStorage.favorite_stations[i])
+				stationDetails(i);
+			}
+		}
+	}
+
+	var stationDetails = function(station_idx) {
+	    $http.get('/station/' + $localStorage.favorite_stations[station_idx])
 					.then(function successCallback(response) {
-					$scope.stations.push(response.data);
+					vm.stations.push(response.data);
 				}, function errorCallback(response) {
 
-			});
-			}
-		}
+		});
 	}
 	
-	$scope.init();
-	
+	init();
```

Comment:
JK...due to the way javascript works...things have to be declared before you use them. leaving as-is